### DISCUSSION TITLE
Type + Mods Attributes with Syntax Errors

### DISCUSSION THREAD

**Discussion by iocron** [mbround18/valheim-docker#596] [General]
Type and Mods in the example https://github.com/mbround18/valheim-docker/blob/main/docs/tutorials/getting_started_with_mods.md are terribly wrong + syntax errors (yaml.parser.ParserError: while parsing a block mapping).

**Comment by spannerman79** [comment]
Unable to replicate.

The mods section is correctly formatted in that example.

In fact my docker-compose has a much larger mods section.

In fact if you outright  copy/paste that example and **only** use it of course you would have issues as you are missing many other dec's (missing mount points, missing vars etc), as its stated **twice** "Example".  **Not** "Use outright as a replacement to use exclusive"

pebkac

**Reply by mbround18** [reply]
Ahh it needed two spaces to be copy+paste-able 

https://github.com/mbround18/valheim-docker/pull/598

**Reply by mbround18** [reply]
@iocron fixed, you can now copy & paste the example. Although, i would highly recommend against trying to persist a server with that config.

**Reply by spannerman79** [reply]
Indeed.

You'll have a server with the same "freedom" as using a GSP (Game Service Provider) if you just outright copy/pasta that docker compose example.

Not recommended or advised.

**Reply by iocron** [reply]
@mbround18 thank you for the feedback/info. If this conf is highly not recommended, then maybe replacing it by a short and full-blown example for productive usage should be considered, if this makes sense? Good examples of any kind do lower the barrier of entry into a topic / software.

**Reply by spannerman79** [reply]
@iocron how long is a peice of string?

You **cannot** give a full blown example as there are people that only want to play vanilla, only want to play with V+, want to play without V+, etc.

It comes down to taking the time to actually read.  If you cannot read then you are out of luck and should use a GSP and let them hand hold you.

Using docker, running your own server - be it a game server, LAMP stack, etc its on **you** to understand what the f- you are doing.

**Reply by iocron** [reply]
@spannerman79 You are quite arrogant aren't you? It's not a problem about reading, because you can always misinterpret something while reading. It's always simpler to explain something based on proper examples, other than explaining something without them. I am using and developing code for docker, docker-compose, self-developed cli's and many other things, but you are forgetting beginners, often the entry is too hard, if you do not provide proper examples and docs. Everyone starts as a beginner somewhere, so did I, others and collegues I know (maybe you are mister universe and know everything, but that's not who everyone is ;-)). If there is a application / use case or language without proper examples, then usually people dodge those most of the time, because you have to waste time trying to figure out something even if you have read everything, while others might be able to explain it better by providing the essential docs and examples.

**Reply by spannerman79** [reply]
> I am using and developing code for docker, docker-compose, self-developed cli's and many other things

Then, feel free to spend the time creating a PR that covers **all** the possible ways that _might_ come up to cover your "scenario". I sure as hell won't.  Hence the statment of "how long is a peice of string?" right at the start.

>You are quite arrogant aren't you?

For stating a fact?  Facts _can_ be hard to swallow, especially when you assume those facts are given with some sort of hidden agenda or plan.  In this case it doesn't apply.  If you want to provide `(1+1n)n` possible outcomes feel free to do so. I'm definitely not going to stop or hold you back.

**Reply by mbround18** [reply]
> @mbround18 thank you for the feedback/info. If this conf is highly not recommended, then maybe replacing it by a short and full-blown example for productive usage should be considered, if this makes sense? Good examples of any kind do lower the barrier of entry into a topic / software.

A production example is available [under the everything + the kitchen sink example](https://github.com/mbround18/valheim-docker#everything-but-the-kitchen-sink) apart from the password and name, that almost identical to what i run in production.
